src/doc/src/reference/resolver.md

```diff
@@ -91,6 +93,31 @@ at the time of this writing) and Package B will use the greatest `0.6` release
 (`0.6.5` for example). This can lead to potential problems, see the
 [Version-incompatibility hazards] section for more details.
 
+The resolver algorithm favors the greatest available versions of dependencies
+and has limited support for backtracking in the solutions it attempts.
+Therefore, it may converge on a solution that includes two copies of a
+dependency when one would suffice. For example:
+
+```toml
+# Package A
+[dependencies]
+rand = "0.7"
+
+# Package B
+[dependencies]
+rand = ">=0.6"
+```
+
+In this example, Cargo may build two copies of the rand crate: `0.8.5` (the
+greatest available at the time of this writing) for Package B and `0.7.3` for
+Package A — even though a single copy at version `0.7.3` meets all requirements.
+Future revisions of the resolver algorithm may change this behavior. In the
```

Comment:
I'm reluctant to imply what might happen in the future, since we don't have hard plans on this. I would probably just leave this sentence out.